### PR TITLE
Handle sensitive optionals coming as null

### LINKED ISSUE #161: Optional sensitive fields are not supported
### What happened?

First reported in https://github.com/crossplane-contrib/provider-tf-equinix-metal/issues/8, fields that are optional and sensitive in the Terraform provider will not prevent resource creation but will cause errors when the resource is being refreshed - thus preventing deletion.

In the Equinix Metal provider's `metal_device` resource, optional sensitive fields `userdata` and `custom_data` are set on device create are not refreshed on subsequent reads. These fields are not forcenew.

https://github.com/equinix/terraform-provider-metal/blob/main/metal/resource_metal_device.go#L268-L282

In Terraform, these fields may appear to be `null` in the `terraform.state` `resources[*].instances[*].attributes` map.

When encountering these values, Terrajet errors out (with a warning): 
```
Warning  CannotObserveExternalResource  30m                  
managed/device.equinixmetal.jet.crossplane.io/v1alpha1, kind=device  
cannot get connection details: cannot get connection details: 
cannot not get a string for fieldpath "user_data": user_data: not a string
```
https://github.com/crossplane-contrib/terrajet/blob/6abd89a444231a61237788434f38d6e92dbae150/pkg/resource/sensitive.go#L119

### How can we reproduce it?

Attempt to create and then delete the following resource: 
https://github.com/crossplane-contrib/provider-tf-equinix-metal/blob/main/examples/device/device.yaml

Or, attempt to delete any resource with an optional sensitive attribute value that has no assigned value.

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

Fixes #161

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

With `device` resource in provider-tf-equinix-metal and also with Unit tests.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/resource/sensitive.go [hunk 1]
```diff
@@ -114,9 +115,18 @@ func GetSensitiveAttributes(from map[string]interface{}, mapping map[string]stri
 		}
 
 		for _, fp := range fieldPaths {
-			v, err := paved.GetString(fp)
+			v, err := paved.GetValue(fp)
 			if err != nil {
```
Comment (muvaf) [inline pkg/resource/sensitive.go:119]: We probably need to ignore `NotFound` error here https://github.com/crossplane/crossplane-runtime/blob/47bff13/pkg/fieldpath/paved.go#L39

File: pkg/resource/sensitive.go [hunk 2]
```diff
@@ -160,6 +170,17 @@ func GetSensitiveParameters(ctx context.Context, client SecretClient, from runti
 			return errors.Wrapf(err, "cannot expand wildcard for xp resource")
 		}
 		for _, expandedJSONPath := range jsonPathSet {
+			v, err := pavedJSON.GetValue(expandedJSONPath)
+			if err != nil {
```
Comment (muvaf) [inline pkg/resource/sensitive.go:174]: Here too https://github.com/crossplane/crossplane-runtime/blob/47bff13/pkg/fieldpath/paved.go#L39

File: pkg/resource/sensitive_test.go [hunk 3]
```diff
@@ -227,6 +231,15 @@ func TestGetSensitiveAttributes(t *testing.T) {
 				},
 			},
 		},
+		"Optional": {
```
Comment (muvaf) [inline pkg/resource/sensitive_test.go:234]: Would you mind adding a case where the field does not exist at all, i.e. not even `null`?

File: pkg/resource/sensitive_test.go [hunk 4]
```diff
@@ -417,13 +426,13 @@ func TestGetSensitiveParameters(t *testing.T) {
...
 						Key: "pass",
 					})).Return([]byte("foo"), nil)
-					client.EXPECT().GetSecretValue(gomock.Any(), gomock.Eq(xpv1.SecretKeySelector{
-						SecretReference: xpv1.SecretReference{
-							Name:      "admin-key",
-							Namespace: "crossplane-system",
-						},
-						Key: "key",
-					})).Return([]byte("bar"), nil)
+					/*					client.EXPECT().GetSecretValue(gomock.Any(), gomock.Eq(xpv1.SecretKeySelector{
```
Comment (muvaf) [inline pkg/resource/sensitive_test.go:429]: Removing these would be cleaner.

**Review by muvaf — COMMENTED** [review]
Thanks @turkenh !

File: pkg/resource/sensitive.go (hunk 1 above)
Comment (muvaf) [reply pkg/resource/sensitive.go:119]: In fact, you may continue to use `GetString` if we ignore the error and check against `""` in the next step, hence remove type casting.

Comment (turkenh) [reply pkg/resource/sensitive.go:119]: @muvaf we have two different cases here:

1. No value at that path (i.e. path does not exist)
2. Path exists but set as `null`/`nil`

First case already covered with https://github.com/crossplane-contrib/terrajet/pull/149 and `NotFound` error is consumed by [expandWildcard](https://github.com/crossplane/crossplane-runtime/blob/d43d510ec5786c455983dede48878b5231de50a8/pkg/fieldpath/paved.go#L191).

This PR is fixing second case where path exists but we are getting a null value.

File: pkg/resource/sensitive.go (hunk 2 above)
Comment (turkenh) [reply pkg/resource/sensitive.go:174]: See response there.

File: pkg/resource/sensitive_test.go (hunk 3 above)
Comment (turkenh) [reply pkg/resource/sensitive_test.go:234]: We already have a test case for that: https://github.com/crossplane-contrib/terrajet/blob/e1a8ff4a3f6c28c31101a95533c54150c1917631/pkg/resource/sensitive_test.go#L230

File: pkg/resource/sensitive.go (hunk 1 above)
Comment (muvaf) [reply pkg/resource/sensitive.go:119]: Got it, I didn't realize we have `null`s in tfstate. Thanks!

**Review by muvaf — APPROVED** [review]
Thanks @turkenh !

**Pull request merged**